src/wgpu_engine.rs

```diff
@@ -130,6 +140,77 @@ impl WgpuEngine {
         }
     }
 
+    /// Enable creating any remaining shaders in parallel
+    pub fn use_parallel_initialisation(&mut self) {
+        if self.shaders_to_initialise.is_some() {
```

Comment:
Scraping the bottom of the barrel a bit, but I suppose that my main question here I guess is since `use_parallel_initialisation` is called immediately after `WgpuEngine::new()`,  and `Vec::new` doesn't allocate until `push`, whether it would be better to just do this initialisation there. 

But came to the realization that since neither `new` or `add_shader` have any hint that things should/should not be conducted in parallel, so to achieve that we would have to add a parameter to `WgpuEngine::new()` unfortunately.
absent that I wonder if it is worth just making this `pub (crate)`, and perhaps also `build_shaders_if_needed`?

it seems fine to disregarding this if there is some reason I don't understand why this should be `pub` or separate from `new`.

Reply:
This isn't actually `pub`, in the sense that `wgpu_engine` isn't externally reachable.

There's one more constraint from future work which might be more enlightening. One of the plans I have to improve startup times further is starting up by using the cpu shaders.
That is, the asynchronous shader compilation I mentioned in the PR description.

However, this would only apply to shaders *other* than fine_area. That is, the call order would be:
```rust
add_shader(fine_area);
use_parallel_initialisation();
add_other_shaders();
build_shaders_if_needed_async();
```

Reply:
Ahh, sorry I noticed that `use wgpu`, but for some reason I read the later `use wgpu::{.., WgpuEngine}` as being `pub`, stand corrected.
I guess that I would still consider moving `use_parallel_initialisation` into `WgpuEngine::new`, but also adding a `parallel` flag to `add_shader` like:

```rust
WgpuEngine::new(...);
add_shader(fine_area, false);
add_other_shaders(); // presumable this calls add_shader(..., true);
build_shaders_if_needed_async();
```

But since this isn't pub as I thought, we aren't actually locking ourselves into any one of these design decisions,
which was my main concern. Because of that i'm happy to resolve this.